SCSS: Add an elevation mixin

shaunandrews opened the pull request:
With this mixin we can add Material-style box-shadows to elements using `@include elevation( 2 )`. You can read more about Material's elevation CSS over here: https://material.io/develop/web/components/elevation/

drw158 commented on assets/stylesheets/shared/mixins/_elevation.scss:
```diff
@@ -0,0 +1,38 @@
+        box-shadow: none
+    }
+    @else if $dp==1 {
+        box-shadow: 0 1px 1px 0 rgba(0, 0, 0, .14), 0 2px 1px -1px rgba(0, 0, 0, .12), 0 1px 3px 0 rgba(0, 0, 0, .20)
```
```suggestion
        box-shadow: 0 1px 1px 0 rgba( var( --color-neutral-900 ), .14), 0 2px 1px -1px rgba( var( --color-neutral-900 ), .12), 0 1px 3px 0 rgba( var( --color-neutral-900 ), .20)
``

This will keep us using variables. We don't currently have a `black` variable but we have a `white` one. Do you think we should have a black one for cases like this?

shaunandrews replied:
Having a `--color-black` seems obvious to me. I honestly expected that we did already.

shaunandrews commented:
Closing this in favor of #31493, which implements Google's elevation npm package, thanks to @scruffian

shaunandrews commented:
Seems like this might just be the easier option after all.

scruffian commented on assets/stylesheets/shared/mixins/_mixins.scss:
```diff
@@ -5,6 +5,7 @@
 @import 'clear-fix';
 @import 'clear-text';
 @import 'dropdown-menu';
+@import 'elevation';
```
As discussed in #31493, I think i'd rather we make this an explicit dependency in the CSS when it's needed, rather than add it to the general utils; this is to make it easier to decouple our components in the future. I'm interested in what @blowery and @jsnajdr think...

jsnajdr replied:
This mixin is going to become rather ubiquitous if we really embrace Material across Calypso, is that right?

At this moment, we rarely do any imports from our `*.scss` files, even these migrated to webpack. The SASS mixins, variables and other constructs that don't output any actual CSS unless used, are implicitly included at the top of each file (see the `prelude` in the `sass-loader` config).

We might want to change that in future and make the imports explicit, but at this moment, the `import 'elevation'` is at the right and expected place.

shaunandrews commented:
Added a new variable for the main color scheme `--color-black` and updated the mixin to use this new variable. I've also pushed a quick example of the mixin in action on signup's `site-type` step.
